chat-widget/src/components/livechatwidget/common/ActivitySubscriber/BotAuthActivitySubscriber.ts

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { BroadcastService } from "@microsoft/omnichannel-chat-components";
+import { BroadcastEvent, LogLevel, TelemetryEvent } from "../../../../common/telemetry/TelemetryConstants";
+import { IActivitySubscriber } from "./IActivitySubscriber";
+import { ICustomEvent } from "@microsoft/omnichannel-chat-components/lib/types/interfaces/ICustomEvent";
+import { TelemetryHelper } from "../../../../common/telemetry/TelemetryHelper";
+const supportedSignInCardContentTypes = ["application/vnd.microsoft.card.signin", "application/vnd.microsoft.card.oauth"];
+const botOauthUrlRegex = /[\S]+.botframework.com\/api\/oauth\/signin\?signin=([\S]+)/;
+const delay = (t: number | undefined) => new Promise(resolve => setTimeout(resolve, t));
+const fetchBotAuthConfigRetries = 3;
+const fetchBotAuthConfigRetryInterval = 1000;
+let response: boolean | undefined;
+
+const extractSignInId = (signInUrl: string) => {
+    const result = botOauthUrlRegex.exec(signInUrl);
+    if (result && result[1]) {
+        return result[1];
+    }
+    return "";
+};
+
+const extractSasUrl = async (attachment: any) => {
+    let sasUrl = undefined;
+    if (attachment && attachment.content && attachment.content.tokenPostResource && attachment.content.tokenPostResource.sasUrl) {
+        sasUrl = attachment.content.tokenPostResource.sasUrl;
+    }
+
+    if (!sasUrl) {
+        const signInId = extractSignInId(attachment.content.buttons[0].value);
+        const getTestUrlEndpoint = `https://token.botframework.com/api/sas/gettesturl?signInId=${signInId}`;
+        try {
+            const response = await (window as any).fetch(getTestUrlEndpoint);
+            if (response.status === 200) {
+                const responseJson = await response.json();
+                sasUrl = responseJson.sasUrl;
+            }
+        } catch {
+            sasUrl = undefined;
+        }
+    }
+    return sasUrl;
+};
+
+const fetchBotAuthConfig = async (retries: number): Promise<any> => {
+    TelemetryHelper.logLoadingEvent(LogLevel.INFO, {
+        Event: TelemetryEvent.SetBotAuthProviderFetchConfig,
+    });
+
+    const executeSigningCardRequest: ICustomEvent = { eventName: BroadcastEvent.ExecuteSigninCardCallbackRequest };
+    BroadcastService.postMessage(executeSigningCardRequest);
+    const listener = BroadcastService.getMessageByEventName(BroadcastEvent.ExecuteSigninCardCallbackResponse)
```

Comment:
Please add a separate documentation instructing consumers to use this feature. Refer to the new doc in the 2 README’s “See more” section at the bottom. The 2 readmes are under root folder and chat-widget folder

Reply:
wip